dbt_subprojects/dex/macros/models/_project/uniswap_compatible_trades.sql

```diff
@@ -118,4 +118,72 @@ SELECT
     , dexs.evt_index
 FROM
     dexs
+{% endmacro %}
+
+{% macro uniswap_compatible_v4_trades(
+    blockchain = null
+    , project = 'uniswap'
+    , version = '4'
+    , Pair_evt_Swap = source('uniswap_v4_sepolia', 'PoolManager_evt_Swap') 
+    , Factory_evt_PoolCreated = source('uniswap_v4_sepolia', 'PoolManager_evt_Initialize')
+    , taker_column_name = 't.evt_tx_from'
+    , maker_column_name = null
+    , optional_columns = []
+    , pair_column_name = 'f.id'
+    )
+%}
+WITH dexs AS
+(
+    SELECT
+        t.evt_block_number AS block_number
+        , t.evt_block_time AS block_time
+        , {{ taker_column_name }} as taker
+        , {% if maker_column_name -%} {{ maker_column_name }} {% else -%} cast(null as varbinary) {% endif -%} as maker      
+        -- in v4, when amount is negative, then user are selling the token (so things are done from the perspective of the user instead of the pool)
+        , CASE WHEN amount0 < INT256 '0' THEN abs(amount1) ELSE abs(amount0) END AS token_bought_amount_raw 
+        , CASE WHEN amount0 < INT256 '0' THEN abs(amount0) ELSE abs(amount1) END AS token_sold_amount_raw
+        , CASE WHEN amount0 < INT256 '0' THEN f.currency1 ELSE f.currency0 END AS token_bought_address
+        , CASE WHEN amount0 < INT256 '0' THEN f.currency0 ELSE f.currency1 END AS token_sold_address
```

Comment:
minor, but can we add column aliases to all columns in the select?
i believe the `amount` fields should have `t.` alias, as they come from swap table

Reply:
added